public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -25,6 +25,9 @@ type Props = {
 };
 
 export function TraceView(props: Props) {
+  if (!props.trace) {
+    return null;
```

Comment:
We should also change the typing somewhere. Probably we need to change the typings here to make it optional and also in parent because it was probably any or non optional there too.

Reply:
Yeah, I was wondering if there is a typing solution to this that would have caught the problem in the first place.

Reply:
Yeah unfortunately we just get it from a dataFrame field so not much type safety there https://github.com/grafana/grafana/blob/a2e6f751c9648ddc2540414cfcf49673e8f7e085/public/app/features/explore/Explore.tsx#L404 but should definitely be something like `Trace | undefined`.